### DISCUSSION TITLE
Is there a bug?

### DISCUSSION THREAD

**Discussion by wddpfx** [eProsima/Fast-DDS#4050] [General]
In file Fast-DDS/src/cpp/fastdds/builtin/typelookup /TypeLookupManager.cpp, there is a function:bool TypeLookupManager::send_reply，
![image_5](https://github.com/eProsima/Fast-DDS/assets/19266720/8eef8c2f-385f-498e-b2f6-76abc9fe8570)
In line 621, maybe there should 
`builtin_reply_writer_history_->remove_change(change);`

**Comment by JLBuenoLopez** [comment]
Thanks for the report, @wddpfx. It is not a bug if you look into the program flux. The function tries to create a new change for the builtin reply writer. If the change is created, the it goes into the condition (`change != nullptr`) and the change is serialized and added into the reply writer history. If the change is not created, for any reason, then the request is removed from the request writer history.

Anyways, the TypeLookup Service is being refactored with the rest of the code related with XTypes v1.3 specification as can be read in [Fast DDS roadmap](https://github.com/eProsima/Fast-DDS/blob/master/roadmap.md#short-term)

**Reply by juanlofer-eprosima** [reply]
Hi @wddpfx , after double checking this we realized it might indeed be a bug. However, as @JLBuenoLopez-eProsima pointed out, we will be tacking this and other identified issues all together in the ongoing refactor.

**Comment by wddpfx** [comment]
好的，收到了你的邮件，我会尽快查看。
